src/Compilers/Core/Portable/Declarations/INamespaceOrTypeDeclaration.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis
+{
+    public interface INamespaceOrTypeDeclaration
+    {
+        string Name { get; }
+
+        ImmutableArray<INamespaceOrTypeDeclaration> Children { get; }
```

Comment:
>Children [](start = 52, length = 8)

maybe this should be moved to `INamespaceDeclaration`, then `ITypeDeclaration.Children` doesn't need to hide this with `new`.

Reply:
If we do that, you lose the way to just triviall enumerate the entire tree without doing any typecasts.  So my preference is to keep this here :)

Reply:
Ok. That does seem convenient :-)

---
In reply to: [267132289](https://github.com/dotnet/roslyn/pull/34067#discussion_r267132289) [](ancestors = 267132289)